metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiRecord.java

```diff
@@ -0,0 +1,42 @@
+package eu.europeana.metis.harvesting.oaipmh;
+
+import eu.europeana.metis.harvesting.HarvesterException;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+/**
+ * This is a immutable class representing an (OAI-embedded) record along with it's OAI header.
+ */
+public class OaiRecord {
+
+  private final OaiRecordHeader header;
+  private final byte[] record;
+
+  /**
+   * Constructor.
+   *
+   * @param header The OAI header of the record.
+   * @param record The record itself (the embedded record).
+   */
+  public OaiRecord(OaiRecordHeader header, byte[] record) {
+    this.header = header;
+    this.record = record;
```

Comment:
Sonar says to store a copy of the array.
We've decided to try and store and retrieve copies of such items.